storage/store/store.go

```diff
@@ -24,6 +24,8 @@ type Store interface {
 	// GetEndpointStatusByKey returns the endpoint status for a given key
 	GetEndpointStatusByKey(key string, params *paging.EndpointStatusParams) (*core.EndpointStatus, error)
 
+	GetGroupAndNameByKey(key string) (string, string, error)
```

Comment:
May not be necessary based on comment mentioning that making a database call to get the key is unnecessary.